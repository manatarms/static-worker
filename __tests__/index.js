import StaticWorker from "../src/index";

let testWorker;

beforeEach(() => {
  testWorker = new StaticWorker(
    new Response("test", {
      url: "myurl.com/images"
    })
  );
});

test("Should build new instance", () => {
  expect(testWorker).toMatchSnapshot();
});

test("Should configure route correctly", () => {
  testWorker.route(/images/, () => {});
  expect(testWorker).toMatchSnapshot();
});

test("Should get response for route", () => {
  testWorker.route(/images/, () => new Response("images response"));
  expect(testWorker.getResponse()).toMatchSnapshot();
});

test("Should return 404 for unregistered routes", () => {
  testWorker.route(/unregistered/, () => new Response());
  expect(testWorker.getResponse()).toMatchSnapshot();
});

test("Should return 404", () => {
  expect(testWorker.notFound()).toMatchSnapshot();
});

test("Should return custom response 404", () => {
  const customResponsetestWorker = new StaticWorker(
    new Response("custom test", {
      url: "custom.com/test"
    }),
    { 404: new Response("Custom 404", { status: 404 }) }
  );
  expect(customResponsetestWorker.notFound()).toMatchSnapshot();
});

test("Should getAssetNameFromUrl", () => {
  expect(testWorker.getResourceNameFromUrl()).toMatchSnapshot();
});

test("Should match first registered route", () => {
  testWorker.route(/images/, () => new Response("images response 1"));
  testWorker.route(/image/, () => new Response("images response 2"));
  expect(testWorker.getResponse()).toMatchSnapshot();
});
