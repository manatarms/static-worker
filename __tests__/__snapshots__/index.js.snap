// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should build new instance 1`] = `
StaticWorker {
  "request": Body {
    "_abort": false,
    "_raw": Array [],
    "body": "test",
    "bodyUsed": false,
    "headers": Headers {
      "_headers": Object {},
    },
    "ok": true,
    "size": 0,
    "status": 200,
    "statusText": "OK",
    "timeout": 0,
    "url": "myurl.com/images",
  },
  "responses": Object {},
  "routes": Map {},
}
`;

exports[`Should configure route correctly 1`] = `
StaticWorker {
  "request": Body {
    "_abort": false,
    "_raw": Array [],
    "body": "test",
    "bodyUsed": false,
    "headers": Headers {
      "_headers": Object {},
    },
    "ok": true,
    "size": 0,
    "status": 200,
    "statusText": "OK",
    "timeout": 0,
    "url": "myurl.com/images",
  },
  "responses": Object {},
  "routes": Map {
    /images/ => [Function],
  },
}
`;

exports[`Should get response for route 1`] = `
Body {
  "_abort": false,
  "_raw": Array [],
  "body": "images response",
  "bodyUsed": false,
  "headers": Headers {
    "_headers": Object {},
  },
  "ok": true,
  "size": 0,
  "status": 200,
  "statusText": "OK",
  "timeout": 0,
  "url": undefined,
}
`;

exports[`Should getAssetNameFromUrl 1`] = `"images"`;

exports[`Should match first registered route 1`] = `
Body {
  "_abort": false,
  "_raw": Array [],
  "body": "images response 1",
  "bodyUsed": false,
  "headers": Headers {
    "_headers": Object {},
  },
  "ok": true,
  "size": 0,
  "status": 200,
  "statusText": "OK",
  "timeout": 0,
  "url": undefined,
}
`;

exports[`Should return 404 1`] = `
Body {
  "_abort": false,
  "_raw": Array [],
  "body": "Not Found",
  "bodyUsed": false,
  "headers": Headers {
    "_headers": Object {},
  },
  "ok": false,
  "size": 0,
  "status": 404,
  "statusText": "Not Found",
  "timeout": 0,
  "url": undefined,
}
`;

exports[`Should return 404 for unregistered routes 1`] = `
Body {
  "_abort": false,
  "_raw": Array [],
  "body": "Not Found",
  "bodyUsed": false,
  "headers": Headers {
    "_headers": Object {},
  },
  "ok": false,
  "size": 0,
  "status": 404,
  "statusText": "Not Found",
  "timeout": 0,
  "url": undefined,
}
`;

exports[`Should return custom response 404 1`] = `
Body {
  "_abort": false,
  "_raw": Array [],
  "body": "Custom 404",
  "bodyUsed": false,
  "headers": Headers {
    "_headers": Object {},
  },
  "ok": false,
  "size": 0,
  "status": 404,
  "statusText": "Not Found",
  "timeout": 0,
  "url": undefined,
}
`;
